wikipedia-sentences/README.md

```diff
@@ -2,103 +2,180 @@
+3. You have at least 8GB of free space on your hard drive. 
+
+
+## Step 1. Clone the repo and install Jina
+
+Begin by cloning the repo so you can get the required files and datasets, and entering the correct folder. 
 ```sh
-curl --request POST -d '{"top_k": 10, "mode": "search",  "data": ["hello world"]}' -H 'Content-Type: application/json' 'http://0.0.0.0:45678/api/search'
+git clone https://github.com/jina-ai/examples
+cd examples/wikipedia-sentences
 ```
 
-## 🐍 Setup
+On your terminal,  you should now be located in you the wikipedia-sentences folder. Let's install Jina and the other required python libraries. For futher information on installing Jina check out [documentation](https://docs.jina.ai/chapters/core/setup/). 
 
 ```sh
 pip install -r requirements.txt
 ```
+If this command runs without any error messages, you can then move onto step two. 
 
-## 📇 Index
-
-We'll start off by indexing a [small dataset of 50 sentences](data/toy-input.txt) to make sure everything is working:
+## Step 2. Indexing your data
+To make life a little easier for you, you can begin by indexing a very [small dataset of 50 sentences](data/toy-input.txt)  to make sure everything is working correctly. 
 
 ```sh
 python app.py -t index
 ```
+If you see the following command, it means your data has been correctly indexed. 
+```sh
+Flow@5162[S]:flow is closed and all resources are released, current build level is 0
+```
+
+We recommend you come back to this step later and index the full wikipedia sentence for better results. 
 
-To index the [full dataset](https://www.kaggle.com/mikeortman/wikipedia-sentences) (almost 900 MB):
+To index the [full dataset](https://www.kaggle.com/mikeortman/wikipedia-sentences) (almost 900 MB) follow these steps:
 
-1. Set up [Kaggle](https://www.kaggle.com/docs/api#getting-started-installation-&-authentication)
+1. Set up a [Kaggle](https://www.kaggle.com/docs/api#getting-started-installation-&-authentication) account
 2. Run the script: `sh ./get_data.sh`
 3. Set the input file: `export JINA_DATA_FILE='data/input.txt'`
 4. Set the number of docs to index `export JINA_MAX_DOCS=30000` (or whatever number you prefer. The default is `50`)
 5. Delete the old index: `rm -rf workspace`
 6. Index your new dataset: `python app.py -t index`
 
-## 🔍 Search
+## Step 3. 🔍 Search your data
+Jina offers several different ways to search (query) your data. In this example, we show three of the most common ones. All three are optional, in a production environment, you would only choose whichever suits your use case best. 
 
-### With REST API
+### Using a REST API
+Begin by running the following command to open the REST API interface.
 
 ```sh
 python app.py -t query_restful
 ```
 
-Then:
+You should open another terminal window and paste the following command. For a quick explanation of what some of these parameter mean,  `top_k`  tells the system how many documents to return. The  `data`  parameter contains the text input you want to query.
```

Comment:
I always prefer to have these params as a table. Also the mode is not quite clear

Reply:
Also 'data' is a list of queries